cmd/operator/internal/controller/kubearchiveconfig_controller.go

```diff
@@ -162,18 +165,28 @@ func (r *KubeArchiveConfigReconciler) reconcileRole(ctx context.Context, kaconfi
 	return role, nil
 }
 
-func (r *KubeArchiveConfigReconciler) desiredRole(kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.Role, error) {
+func (r *KubeArchiveConfigReconciler) desiredRole(ctx context.Context, kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.Role, error) {
+	log := log.FromContext(ctx)
 	var rules []rbacv1.PolicyRule
 	for _, resource := range kaconfig.Spec.Resources {
 		apiGroup := ""
-		slash := strings.Index(resource.APIVersion, "/")
-		if slash > -1 {
-			apiGroup = resource.APIVersion[:slash]
+		apiVersion := resource.APIVersion
+		data := strings.Split(resource.APIVersion, "/")
+		if len(data) > 1 {
+			apiGroup = data[0]
+			apiVersion = data[1]
+		}
+		// The resource field in the GVR contains the plural version of the resource, and
+		// the kubernetes Role expects this lower-cased plural version.
+		gvr, err := r.Mapper.RESTMapping(schema.GroupKind{Group: apiGroup, Kind: resource.Kind}, apiVersion)
+		if err == nil {
+			rules = append(rules, rbacv1.PolicyRule{
+				APIGroups: []string{apiGroup},
+				Resources: []string{strings.ToLower(gvr.Resource.Resource)},
+				Verbs:     []string{"get", "list", "watch"}})
+		} else {
+			log.Error(err, "Failed to get GVR for "+resource.APIVersion)
```

Comment:
WDYT about this: `log.Error("Failed to get GVR for: %s", resource.APIVersion, err)`

Reply:
`[log.Error](https://pkg.go.dev/github.com/go-logr/logr#Logger.Error)` does not support format verbs, like `%s` and `logr.Logger` does not appear to have an `Errorf` method

Reply:
This PR has been merged already.